simphony/testing/abc_check_engine.py

```diff
@@ -212,6 +215,7 @@ def test_dataset_rename(self):
         # we should be able to access using the new "bar" name
         ds_bar = engine.get_dataset("bar")
         self.assertEqual(ds_bar.name, "bar")
+        engine.remove_dataset("bar")
```

Comment:
Yeah, the test is about renaming these proxy datasets.  After we rename the dataset from "foo" to "bar", we do some testing that can be broken in 3 parts:

 * 1st part: Then we test that we can't get a "foo" named database (cause its been renamed to "bar").  
 * 2nd part:  we can use the "bar" name
 * 3rd part:  finally, we test if we can add a new dataset called "foo".  We should be allowed to do this as there is only a "bar" dataset at the moment. 

**Only** this final step ("3rd part") is there a problem for wrappers that only allow a single dataset as they can't add a second dataset to test this behaviour.  So that is why I am proposing to add the if-statement there to skip this 3rd part for single-dataset wrappers.

-------------------

If we use 1ad0dfb , there is a problem as we are are deleting the "bar" dataset which weakens the point of the 3rd part of this step.  If we leave this commit (1ad0dfb), the 3rd part ends up just testing if we can add a dataset to wrapper which has no datasets.  This doesnt match up with the original intention of testing if we can add a data set with the 'no-longer used "foo" name`